projects/hathor-wallet-headless/hsm-c-bridge.md

```diff
@@ -0,0 +1,327 @@
+- Feature Name: hsm_c_bridge
+- Start Date: 2023-07-31
+- RFC PR:
+- Hathor Issue:
+- Author: Tulio Vieira tulio@hathor.network
+
+# Summary
+[summary]: #summary
+
+This document describes the C Bridge between the Headless Wallet and the Dinamo Networks Hardware Security System (from [Dinamo Networks](https://www.dinamonetworks.com/en/hardware-security-module-hsm/)) ( HSM ).
+
+# Motivation
+[motivation]: #motivation
+
+This bridge is necessary to make NodeJS [client applications](https://github.com/HathorNetwork/rfcs/blob/feat/master/projects/hathor-wallet-headless/hsm-integration.md) communicate with HSM devices while there is no official SDK from Dinamo for the javascript language.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The bridge is comprised mostly of code from the [examples on Dinamo docs site](https://manual.dinamonetworks.io/c/examples.html), with added structures to facilitate communication with the NodeJS through `stdin` input and output interpretation of the `stdout` and/or `stderr`.
```

Comment:
Since the main designed usage will be on the headless with docker, maybe we should not start a secondary process to communicate with HSM.
A known rule of thumb in docker is to have a single process per container, we are not locked from starting other processes but we should avoid where possible ([reference](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#decouple-applications))

Reply:
Good points! They were added in the _Drawbacks_ and _Alternatives_ sections.